src/metabase/api/x_ray.clj

```diff
@@ -32,163 +33,133 @@
    {:query       (keyword query)
     :computation (keyword computation)}))
 
-(api/defendpoint GET "/field/:id"
-  "Get x-ray for a `Field` with ID."
-  [id max_query_cost max_computation_cost]
+(def ^:private ->model
+  {"field"    Field
+   "segment"  Segment
+   "table"    Table
+   "card"     Card
+   "segments" Segment
+   "tables"   Table
+   "cards"    Card
+   "fields"   Field})
+
+(def ^:private Model
+  (apply s/enum (keys ->model)))
+
+(api/defendpoint GET "/:model/:id"
+  "Get x-ray for model of type `:model` with id `:id`."
+  [id model max_query_cost max_computation_cost]
   {max_query_cost       MaxQueryCost
-   max_computation_cost MaxComputationCost}
-  (api/check-403 (costs/enable-xrays))
-  (->> id
-       (api/read-check Field)
-       (fe/extract-features {:max-cost (max-cost max_query_cost
-                                                 max_computation_cost)})
-       fe/x-ray))
-
-(api/defendpoint GET "/table/:id"
-  "Get x-ray for a `Tield` with ID."
-  [id max_query_cost max_computation_cost]
-  {max_query_cost       MaxQueryCost
-   max_computation_cost MaxComputationCost}
+   max_computation_cost MaxComputationCost
+   model                Model}
   (api/check-403 (costs/enable-xrays))
   (->> id
-       (api/read-check Table)
+       (api/read-check (->model model))
        (fe/extract-features {:max-cost (max-cost max_query_cost
                                                  max_computation_cost)})
        fe/x-ray))
 
-(api/defendpoint GET "/segment/:id"
-  "Get x-ray for a `Segment` with ID."
-  [id max_query_cost max_computation_cost]
-  {max_query_cost       MaxQueryCost
-   max_computation_cost MaxComputationCost}
-  (api/check-403 (costs/enable-xrays))
-  (->> id
-       (api/read-check Segment)
-       (fe/extract-features {:max-cost (max-cost max_query_cost
-                                                 max_computation_cost)})
-       fe/x-ray))
+(defn- virtual-card
```

Comment:
why is this called a `virtual-card`? It's just a query that returns a result. The use of `card` is an anachronism we shouldn't hold on to.

Reply:
Is the models/Query the correct model to use here? It would be handy if it's wrapped in some type so I can dispatch on it.